zerver/tests/test_subs.py

```diff
@@ -1925,34 +1925,41 @@ def test_unarchive_stream_private_and_web_public(self) -> None:
             acting_user=zoe,
         )
         self.subscribe(self.example_user("cordelia"), "stream_private_name1")
-        with self.capture_send_event_calls(expected_num_events=2) as events:
+        with self.capture_send_event_calls(expected_num_events=3) as events:
             do_unarchive_stream(stream, new_name="private", acting_user=None)
 
         stream = Stream.objects.get(id=stream.id)
         self.assertFalse(stream.is_web_public)
 
         # Tell all users with metadata access that the stream exists.
-        self.assertEqual(events[0]["event"]["op"], "create")
-        self.assertEqual(events[0]["event"]["streams"][0]["name"], "private")
-        self.assertEqual(events[0]["event"]["streams"][0]["stream_id"], stream.id)
-        notified_user_ids = set(events[0]["users"])
-        self.assertEqual(
-            notified_user_ids,
-            can_access_stream_metadata_user_ids(stream),
-        )
-        self.assertIn(self.example_user("cordelia").id, notified_user_ids)
-        # An important corner case is that all organization admins are notified.
-        self.assertIn(self.example_user("iago").id, notified_user_ids)
-        # The current user, Hamlet was made an admin and thus should be notified too.
-        self.assertIn(aaron.id, notified_user_ids)
-        # Channel admin should be notified.
-        self.assertIn(self.example_user("aaron").id, notified_user_ids)
-        # User belonging to `can_add_subscribers_group` should be notified.
-        self.assertIn(prospero.id, notified_user_ids)
-        # User belonging to `can_subscribe_group` should be notified.
-        self.assertIn(zoe.id, notified_user_ids)
-        # Guest user should not be notified.
-        self.assertNotIn(self.example_user("polonius").id, notified_user_ids)
+        self.assertEqual(events[0]["event"]["op"], "update")
+        self.assertEqual(events[0]["event"]["stream_id"], stream.id)
+        self.assertEqual(events[0]["event"]["property"], "is_archived")
+        self.assertEqual(events[0]["event"]["value"], False)
+
+        self.assertEqual(events[1]["event"]["op"], "create")
+        self.assertEqual(events[1]["event"]["streams"][0]["name"], "private")
+        self.assertEqual(events[1]["event"]["streams"][0]["stream_id"], stream.id)
```

Comment:
Probably worth adding a comment that clients should only get one of these depending on whether they support the `archived_channels` client capability?

Reply:
Added comments.